pkg/agent/util/winnet/interfaces.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package winnet
+
+import "antrea.io/antrea/pkg/agent/util"
+
+type Interface interface {
```

Comment:
We're back to the same situation as before (https://github.com/antrea-io/antrea/pull/5159#discussion_r1330745265).
Is it not possible to also have the implementation (real implementation, not generate mock) of the interface in the `winnet` package?

Reply:
It should be possible, but it may make many changes to current code. We should move the real implementation to `winnet` package. We also need reserve some functions that have the same name but have different implementations in `net_windows.go` and `net_linux.go`, for example, like `func SetLinkUp(name string) (net.HardwareAddr, int, error)`, which be used by both Windows and Linux. Maybe should we should do this in another PR.

Reply:
But I just have in mind the functions from this interface. These are specific to Windows, and not shared?

Reply:
You are right, these functions are specific to Windows. I have tried this:

-  Move those real implementations including `RouteListFiltered` to package `winnet`
-  Don't move the structs like `Route`, `Neighbor`, `NetNatStaticMapping` to package `winnet`. 
- There are some functions like `GetInterfaceConfig` which should be left on the original file using `RouteListFiltered` which will be in package `winnet`. The reason why this function should be left on the original file is that the function will be used on both Linux and Windows. 

From above, structs like `Route` defined in the original package is used in package `winnet`, but function `RouteListFiltered` in package `winnet` is also used in the original package. This results import cycling. 

Another simple way is that we move all code related to Windows to package `winnet`, but we still still need reserve some functions in the original packet like `RenameInterface`, `SetInterfaceMTU`, etc since these function are used on both Linux and Windows. Another issue introduced is that we need to refactor the test code.

Anyway, it is a little complicated to complete the moving of these code. How about refactoring this in another PR?